mlir/lib/Dialect/Math/Transforms/ExpandPatterns.cpp

```diff
@@ -311,40 +311,83 @@ static LogicalResult convertFPowIOp(math::FPowIOp op,
   return success();
 }
 
-// Converts  Powf(float a, float b) (meaning a^b) to exp^(b * ln(a))
+// Convert Powf(float a, float b) for special cases when b is constant:
+// when b == 0, or |b| == 0.5, 1.0, or 2.0.
+static LogicalResult convertSpecialPowfOp(math::PowFOp op,
+                                          PatternRewriter &rewriter) {
+  ImplicitLocOpBuilder b(op->getLoc(), rewriter);
+  Value operandA = op.getOperand(0);
+  Value operandB = op.getOperand(1);
+  auto opType = operandA.getType();
+  auto baseType = operandB.getType();
+
+  auto &sem = dyn_cast<mlir::FloatType>(getElementTypeOrSelf(baseType))
```

Comment:
Here, since `math.powf` requires float arguments (I just checked MathOps.td), the cast really shouldn't ever fail, so I think you can simply use `cast` instead of `dyn_cast`.  You weren't checking for a null return value from `dyn_cast` anyway.

Reply:
Done!